### PR TITLE
Introduce TkSync.Sync to automate unread threads notifications

### PR DESCRIPTION
**Pull request by victornoel**
This is for #29 

In order to automate unread threads notification, I introduced the `TkSync.Sync` interface that has multiple implementations:
- previously existing Threads updates
- unread threads notifications

I removed the `Notification` abstraction that didn't seem to serve any purpose now.

`@todo`s:
- one for continuing the work to actually instantiate `UnreadThreadsSync`.
- one for improving `TkSync` behaviour now that it can execute multiple tasks
- I moved some `@todo`s but didn't change their content: 0pdd should recognize those without problem.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#66](https://github.com/g4s8/ghman/issues/66) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#66](https://github.com/g4s8/ghman/issues/66) is assigned to @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova), here is [why](https://www.0crat.com/footprint/CJT4R32Q3/cb50718a-5fc3-4124-abbb-53c149263551); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/com/g4s8/ghman/web/TkSync.java [hunk 1]
```diff
@@ -16,51 +16,52 @@
...
+            sync.sync();
         }
         return new RsEmpty();
     }
+
+    /**
+     * Sync.
+     * @since 1.0
+     */
+    public interface Sync {
```
Comment (olenagerasimova) [inline src/main/java/com/g4s8/ghman/web/TkSync.java]: @victornoel do not you mind to extract this interface into separate file? Naming `TkSync.Sync` seems off

File: src/main/java/com/g4s8/ghman/user/PgThreads.java [hunk 2]
```diff
@@ -37,7 +37,7 @@
  * @since 1.0
  * @checkstyle MagicNumberCheck (500 lines)
```
Comment (olenagerasimova) [inline src/main/java/com/g4s8/ghman/user/PgThreads.java]: @victornoel I'd suggest to move this checkstyle suppression to methods scope, it seems to be a good practice to make suppressions work for as small peace of code as possible

**Comment by olenagerasimova** [comment]
@victornoel ping

**Comment by victornoel** [comment]
@olenagerasimova here it is

**Comment by olenagerasimova** [comment]
@rultor merge

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
@ammaratef45[/z](https://www.0crat.com/u/ammaratef45) please review this job completed by @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Job [#66](https://github.com/g4s8/ghman/issues/66) is not in the agenda of @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova), can't inspect

**Comment by 0crat** [comment]
The job [#66](https://github.com/g4s8/ghman/issues/66) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ammaratef45** [comment]
@olenagerasimova You didn't manage to find three or more problems with the PR

please confirm you will try to find 3 or more problems next time

**Comment by olenagerasimova** [comment]
@ammaratef45 I confirm

**Comment by ammaratef45** [comment]
@olenagerasimova Thanks

**Comment by ammaratef45** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @ammaratef45[/z](https://www.0crat.com/u/ammaratef45)

**Pull request merged**